### DISCUSSION TITLE
ImportError: cannot import name '_cffirmware' from 'pycrazyswarm.cfsim'

### DISCUSSION THREAD

**Discussion by ElaineYao** [USC-ACTLab/crazyswarm#628] [Q&A] [answered]
I'm following the [guideline](https://crazyswarm.readthedocs.io/en/latest/installation.html) for physical robots and simulation.
```
$ cd ros_ws/src/crazyswarm/scripts
$ source ../../../devel/setup.bash
$ $CSW_PYTHON -m pytest
```
After I run `$CSW_PYTHON -m pytest`, it gave me the error `ImportError: cannot import name '_cffirmware' from 'pycrazyswarm.cfsim'`

Plus, since my laptop is Ubuntu 20.04 with python 3.8 and ROS Noetic, in the `build.sh`, I add one line to compile it.
```
# ros
cd ros_ws
catkin_make -DPYTHON_EXECUTABLE=/usr/bin/python3  # I add this
catkin_make -DCMAKE_BUILD_TYPE=RelWithDebInfo
cd $ROOT
```

**Comment by jpreiss** [comment, marked as answer]
this step in `build.sh` must have failed: https://github.com/USC-ACTLab/crazyswarm/blob/967723df9dc1e27bae1c2725d46a182c8f3c2f80/build.sh#L20-L23
